pydantic_ai_slim/pydantic_ai/ext/temporal/_agent.py

```diff
@@ -0,0 +1,330 @@
+
+        def temporalize_toolset(toolset: AbstractToolset[AgentDepsT]) -> AbstractToolset[AgentDepsT]:
+            id = toolset.id
+            if not id:
+                raise UserError(
+                    "Toolsets that implement their own tool calling need to have an ID in order to be used with Temporal. The ID will be used to identify the toolset's activities within the workflow."
+                )
+            toolset = temporalize_toolset_func(
+                toolset,
+                activity_config | toolset_activity_config.get(id, {}),
+                tool_activity_config.get(id, {}),
+                run_context_type,
+            )
+            if isinstance(toolset, TemporalWrapperToolset):
+                activities.extend(toolset.temporal_activities)
+            return toolset
+
+        # TODO: Use public methods so others can replicate this
+        temporal_toolsets = [
+            temporalize_toolset(toolset)
+            for toolset in [agent._function_toolset, *agent._user_toolsets]  # pyright: ignore[reportPrivateUsage]
+        ]
+
+        self._model = temporal_model
+        self._toolsets = temporal_toolsets
+        self._temporal_activities = activities
+
+    @property
+    def model(self) -> Model:
+        return self._model
+
+    @property
+    def temporal_activities(self) -> list[Callable[..., Any]]:
+        return self._temporal_activities
+
+    @overload
+    def iter(
+        self,
+        user_prompt: str | Sequence[_messages.UserContent] | None = None,
+        *,
+        output_type: None = None,
+        message_history: list[_messages.ModelMessage] | None = None,
+        model: models.Model | models.KnownModelName | str | None = None,
+        deps: AgentDepsT = None,
+        model_settings: ModelSettings | None = None,
+        usage_limits: _usage.UsageLimits | None = None,
+        usage: _usage.Usage | None = None,
+        infer_name: bool = True,
+        toolsets: Sequence[AbstractToolset[AgentDepsT]] | None = None,
+        **_deprecated_kwargs: Never,
+    ) -> AbstractAsyncContextManager[AgentRun[AgentDepsT, OutputDataT]]: ...
+
+    @overload
+    def iter(
+        self,
+        user_prompt: str | Sequence[_messages.UserContent] | None = None,
+        *,
+        output_type: OutputSpec[RunOutputDataT],
+        message_history: list[_messages.ModelMessage] | None = None,
+        model: models.Model | models.KnownModelName | str | None = None,
+        deps: AgentDepsT = None,
+        model_settings: ModelSettings | None = None,
+        usage_limits: _usage.UsageLimits | None = None,
+        usage: _usage.Usage | None = None,
+        infer_name: bool = True,
+        toolsets: Sequence[AbstractToolset[AgentDepsT]] | None = None,
+        **_deprecated_kwargs: Never,
+    ) -> AbstractAsyncContextManager[AgentRun[AgentDepsT, RunOutputDataT]]: ...
+
+    @overload
+    @deprecated('`result_type` is deprecated, use `output_type` instead.')
```

Comment:
I feel like we can drop these deprecated signatures here

Reply:
Like, we can even still keep it working by keeping the implementation the same and handling the deprecated arguments, just remove it from type-checking.